File: src/Poker/hand/StraightFlush.java
package Poker.hand;

import Poker.game.DeckConstants.Value;

public class StraightFlush extends Straight
{   
   public StraightFlush(Value high)
   {
      super(high);
      m_oRank = Rank.STRAIGHT_FLUSH;
   }
   
   public String toString()
   {
      return "Straight flush - " + high.getIndex() + " high";
   }
}

File: server/hand/straightFlush.js
let Hand = require('./hand');

let StraightFlush = (highValue) =>
{
	const STRAIGHT_FLUSH = 8;
	
	let rank = Hand(STRAIGHT_FLUSH);
	
	let self =
	{
		getRank: () => rank,
		compare: (hand) =>
		{
			let rankCompare = rank.compare(hand.getRank());
			if (rankCompare)
			{
				return rankCompare;
			}
			
			return self.compareSameRank(hand);
		},
		compareSameRank: (straightFlush) =>
		{
			return highValue - straightFlush.getHighValue();
		},
		getHighValue: () => highValue,
		toString: () => 'Straight flush (' + highValue + ' high)'
	}
	
	return self;
}

module.exports = StraightFlush;
